src/utils/utils.ts

```diff
@@ -141,14 +152,52 @@ export async function setupGitIgnore(folderPath: string): Promise<void> {
       process.logger?.success('Existing .gitignore has been updated.')
     }
   } else {
-    shelljs.exec(`cd ${folderPath} && git init`, {
+    await createFile(
+      gitIgnoreFilePath,
+      'node_modules/\nsasjsbuild/\nsasjsresults/\n.env\n'
+    )
+
+    process.logger?.success('.gitignore file has been created.')
+  }
+
+  const folderPath = path.join(process.projectDir, folderName)
+  const gitDirectoryExists = await folderExists(path.join(folderPath, '.git'))
+  if (!gitDirectoryExists) {
+    shelljs.exec(`cd ${folderName} && git init`, {
       silent: true
     })
+  }
+}
 
-    await createFile(gitIgnoreFilePath, 'node_modules/\nsasjsbuild/\n.env\n')
+export async function setupGhooks(folderName: string) {
+  const folderPath = path.join(process.projectDir, folderName)
 
-    process.logger?.success('.gitignore file has been created.')
-  }
+  process.logger?.info('Installing ghooks')
+  shelljs.exec(`cd "${folderPath}" && npm i ghooks --save-dev`, {
+    silent: true
+  })
+
+  try {
+    const packageJsonPath = path.join(folderPath, 'package.json')
+    const packageJsonContent = await readFile(packageJsonPath)
+    const packageJson = JSON.parse(packageJsonContent)
+
+    if (!packageJson.config) packageJson.config = {}
+
+    if (!packageJson.config.ghooks) packageJson.config.ghooks = {}
+
+    let preCommitCmd = 'sasjs lint'
+    if (packageJson.config.ghooks['pre-commit']) {
```

Comment:
we need a test to ensure `sasjs lint` is not already in the command here, right?